demisto_sdk/commands/find_dependencies/find_dependencies.py

```diff
@@ -275,99 +299,153 @@ def _collect_scripts_dependencies(pack_scripts, id_set):
                 pack_names = PackDependencies._search_packs_by_integration_command(command, id_set)
 
                 if pack_names:  # found integration dependency implementing pack name
                     pack_dependencies_data = PackDependencies._detect_generic_commands_dependencies(pack_names)
-                    dependencies_packs.update(pack_dependencies_data)
+                    script_dep.update(pack_dependencies_data)
+
+            verbose_file.write(
+                f'{os.path.basename(script.get("file_path", ""))} depends on: '
+                f'{script_dep}  '
+            )
+            dependencies_packs.update(script_dep)
 
         return dependencies_packs
 
     @staticmethod
-    def _collect_playbooks_dependencies(pack_playbooks, id_set):
+    def _differentiate_playbook_implementing_objects(implementing_objects, skippable_tasks, id_set_section):
+        """
+        Differentiate implementing objects by skippable.
+
+        Args:
+            implementing_objects (list): playbook object collection.
+            skippable_tasks (set): playbook skippable tasks.
+            id_set_section (list): id set section corresponds to implementing_objects (scripts or playbooks).
+
+        Returns:
+            set: dependencies data that includes pack id and whether is mandatory or not.
+        """
+        dependencies = set()
+
+        mandatory_scripts = set(implementing_objects) - skippable_tasks
+        optional_scripts = set(implementing_objects) - mandatory_scripts
+
+        optional_script_packs = PackDependencies._search_packs_by_items_names(list(optional_scripts),
+                                                                              id_set_section)
+        if optional_script_packs:  # found packs of optional objects
+            pack_dependencies_data = PackDependencies._label_as_optional(optional_script_packs)
+            dependencies.update(pack_dependencies_data)
+
+        mandatory_script_packs = PackDependencies._search_packs_by_items_names(list(mandatory_scripts),
+                                                                               id_set_section)
+        if mandatory_script_packs:  # found packs of mandatory objects
+            pack_dependencies_data = PackDependencies._label_as_mandatory(mandatory_script_packs)
+            dependencies.update(pack_dependencies_data)
+
+        return dependencies
+
+    @staticmethod
+    def _collect_playbooks_dependencies(pack_playbooks, id_set, verbose_file):
         """
         Collects playbook pack dependencies.
 
         Args:
             pack_playbooks (list): collection of pack playbooks data.
             id_set (dict): id set json.
+            verbose_file (VerboseFile): path to dependency explanations file.
 
         Returns:
             set: dependencies data that includes pack id and whether is mandatory or not.
 
         """
         dependencies_packs = set()
+        verbose_file.write('\n### Playbooks')
```

Comment:
Good idea. I think in later step the log should be extended and contain the reason and additional data for found dependency.

Reply:
what do you mean later step?
this is output example:

![image](https://user-images.githubusercontent.com/30797606/87878721-a7a18000-c9ee-11ea-8188-d19daac829c6.png)

Reply:
I mean that if layout from pack A uses specific field from pack B then better to log it somewhere. But it is not urgent.
For example: https://panw-global.slack.com/archives/G011E63JXPB/p1594907099182700